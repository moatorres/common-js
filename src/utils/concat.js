function concat(original = [], extension = []) {
  let index
  let result = []

  index = 0

  while (index < original.length) {
    result[result.length] = original[index]
    index += 1
  }

  index = 0

  while (index < extension.length) {
    result[result.length] = extension[index]
    index += 1
  }

  return result
}

export default concat
