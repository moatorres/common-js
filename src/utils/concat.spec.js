import concat from './concat'

describe('concat()', () => {
  it('Should be defined', () => {
    expect(concat).toBeDefined()
  })

  it('Should concatenate two arrays', () => {
    let a = [1, 2, 3]
    let b = [4, 5, 6]
    let res = concat(a, b)

    expect(res).toMatchSnapshot()
    expect(res).toIncludeAllMembers(a)
    expect(res).toIncludeAllMembers(b)
  })

  it('Should be able to concatenate two strings', () => {
    let a = 'Hey, '
    let b = 'buddy!'

    // @ts-ignore
    let res = concat(a, b).join('')
    expect(res).toEqual('Hey, buddy!')
  })
})
