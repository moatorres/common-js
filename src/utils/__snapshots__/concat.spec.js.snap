// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`concat() Should concatenate two arrays 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
  6,
]
`;
